internal/controller/k8s_resources.go

```diff
@@ -126,6 +129,33 @@ func convertMetaObjectToMetadata(obj metav1.ObjectMeta, resource string) *pb.Kub
 	return objMetadata
 }
 
+// getPodIPAddresses uses a pod name and namespace to grab the hostIP addresses within the podStatus
+func getPodIPAddresses(ctx context.Context, logger *zap.SugaredLogger, podName string, namespace string) ([]v1.HostIP, error) {
```

Comment:
Why are you re-querying the Pod?
You already receive the full Pod object from the watcher.
No need to re-query data you already have.